### PR TITLE
Cast equivalent float64s to int64 rather than failing with ErrInvalidValue

### PR DESCRIPTION
**Pull request by abmyii**
When working with Pandas (both with and without `doltpy`) I ran into this issue multiple times - integer columns are converted to floats (due to the way Python handles `nan` values), which then causes `dolt table import` to complain with the following error despite the floats being "integral":
```
Rows Processed: 0, Additions: 0, Modifications: 0, Had No Effect: 0

A bad row was encountered while moving data.
Bad Row: 
error: '10.0' is not a valid value for 'INT'
These can be ignored using the '--continue'
```
This PR converts "integral"/equivalent floats to int64s to prevent this from happening. This still prevents non-integral floats from being imported, e.g.:
```
Rows Processed: 0, Additions: 0, Modifications: 0, Had No Effect: 0

A bad row was encountered while moving data.
Bad Row: 
error: '10.1' is not a valid value for 'INT'
These can be ignored using the '--continue'
```

I'm not sure how/where this should be tested, but if it is an acceptable PR I'll be happy to write the tests for it too.

P.S. The `isIntegral` function can be removed and used in the if-statement as a condition if that's more preferable, though I think  it should be documented (perhaps in a comment) since it's purpose may not be immediately obvious.

### CODE REVIEW

**Review by zachmu — COMMENTED** [review]
Think I'm fine with this, but it needs tests, both for this unit and the engine:

numbertypes_test.go
insert_queries.go

**Comment by abmyii** [comment]
> Think I'm fine with this, but it needs tests, both for this unit and the engine:
> 
> numbertypes_test.go insert_queries.go

OK, will do. Is my last commit (running `format_repo.sh`) OK or should I ignore that? It doesn't seem to have been run on most of the code.

**Comment by zachmu** [comment]
I would back your formatting commit out of your branch and re-push. It looks like you're using a slightly different version of go fmt than our CI is, so it's introducing a lot of differences.

I can reformat the code once you have tests in.

**Comment by abmyii** [comment]
I'm not sure how to run the tests - could you please detail the commands?

**Comment by timsehn** [comment]
https://www.dolthub.com/blog/2021-09-08-how-to-contribute-to-dolt/

From there:

`~/dolt_workspace/dolt/go $ go test ./...`

**Comment by abmyii** [comment]
> `~/dolt_workspace/dolt/go $ go test ./...`

Thanks! I ran this command before but I wasn't sure which tests it was running exactly.

**Comment by abmyii** [comment]
All the tests passed apart from `error: '9223372036854775807' is not a valid value for 'BIGINT'` which occurred even before my changes were made.

File: sql/numbertype.go [hunk 1]
```diff
@@ -509,11 +513,11 @@ func convertToInt64(t numberTypeImpl, v interface{}) (int64, error) {
 		}
 		return i, nil
 	case string:
-		i, err := strconv.ParseInt(v, 10, 64)
-		if err != nil {
+		f, err := strconv.ParseFloat(v, 64)
+		if err != nil || !isIntegral(f) {
```
Comment (zachmu) [inline sql/numbertype.go:517]: See other comment, this is actually not an appropriate check.

**Review by zachmu — COMMENTED** [review]
Looks good, but the error behavior on non-integral floats is actually wrong. Fix that and we're good to merge.

**Comment by abmyii** [comment]
@zachmu Fixed! I would prefer if it didn't accept non-integral floats/gave a warning when truncating, but oh well!

**Review by zachmu — APPROVED** [review]
LGTM, thanks for the contribution

**Comment by zachmu** [comment]
Replaced with #630

**Pull request closed without merging**